[FLINK-22352][mesos] Deprecates Mesos support (#15673)
apache/flink: docs/content.zh/docs/deployment/config.md

### Mesos

{{< hint warning >}}

> Apache Mesos support was deprecated in Flink 1.13 and will become subject of removal in the future (see 
> [FLINK-22352](https://issues.apache.org/jira/browse/FLINK-22352) for further details).

Reviewer: ```suggestion
Apache Mesos support was deprecated in Flink 1.13 and is subject to removal in the future (see 
[FLINK-22352](https://issues.apache.org/jira/browse/FLINK-22352) for further details).
```

Author: Thanks @zentol for your review. I applied your change request on all affected sources.
